packages/components/src/ui/create-styles/create-style-system/create-core-element.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * External dependencies
+ */
+import isPropValid from '@emotion/is-prop-valid';
+
+/**
+ * WordPress dependencies
+ */
+import { useMergeRefs } from '@wordpress/compose';
+import { forwardRef, createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useHydrateGlobalStyles } from '../hooks';
+import {
+	INTERPOLATION_CLASS_NAME,
+	REDUCED_MOTION_MODE_ATTR,
+} from './constants';
+import {
+	DEFAULT_STYLE_SYSTEM_OPTIONS,
+	getInterpolatedClassName,
+} from './utils';
+
+const shouldForwardProp = isPropValid;
+
+const defaultOptions = DEFAULT_STYLE_SYSTEM_OPTIONS;
+
+/**
+ * @typedef CreateCoreElementOptions
+ * @property {import('create-emotion').ObjectInterpolation<any>} baseStyles The baseStyles from the Style system.
+ * @property {import('../create-compiler').Compiler} compiler The injectGlobal from the Style system's compiler.
+ * @property {import('./generate-theme').GenerateThemeResults} globalStyles The globalStyles from the Style system.
+ */
+
+/**
+ * Creates the core styled elements for the Style system. These elements are
+ * an in-between of Emotion's <div css="" /> solution and a styled.div``
+ * solution.
+ *
+ * createCoreElement is a super light-weight higher-order wrapper that
+ * enhances base elements (like `div`, `input`, or even Components) with
+ * features provided by the Style system, such as direct CSS compiling via
+ * the `css` prop.
```

Comment:
Can we remove this?

Reply:
Just remove the documentation or support for the `css` prop? The `css` prop is used by the component system internally (and it's a powerful tool that `emotion` exposes itself as well). Even if we decide not to have it be a documented external API I think we should leave documentation about why `createCoreElement` exists, right? Otherwise it will be a mystery even to those contributing to `@wordpress/components` what this `css` prop is or where it comes from for `styled` components.

Reply:
I thought (and read somewhere in the docs here) that internally the styled components work by just supporting the "className" prop, isn't that true? In which cases, we do need the css prop? I'm asking cause if we allow it in some components but not others, it might be harder to distinguish between "internal" or "public" components.

Reply:
So to be entirely honest up until now I hadn't done an audit of the `css` prop's usage, but it's actually super minimal outside of stories and tests in G2.

I believe we could remove it altogether with minimal disruption (we'll have to re-write some stories as they come in but that's not a big deal).

The `css` prop is merely a convenience prop that allows you to pass styles inline, similar to the `style` prop but more powerful because it leverages all of the style system in there. However, you can basically do the same thing with the `className` prop so I'm not sure I'm convinced the `css` prop is actually a necessity (it might even be an anit-pattern like inline styles tend to be).